### PR TITLE
Fix search tap

### PR DESCRIPTION
**Pull request by davidfokkema**
I've reimplemented `brew search <tap>` and `brew search <tap> query`.

### CODE REVIEW

**Comment by bfontaine** [comment]
Any update on this?

**Comment by davidfokkema** [comment]
I still like it. As far as I can tell, it is still not possible to list all formulae from a particular (untapped) tap, and it is not possible to search only inside a particular tap. But after one year of development, this PR now has conflicts. I'm willing to look a it again, but only if there is interest from the maintainers to actually merge this.

File: Library/Contributions/manpages/brew.1.md [hunk 1]
```diff
@@ -305,10 +305,15 @@ Note that these flags should only appear after a command.
     Display all locally available formulae for brewing (including tapped ones).
     No online search is performed if called without arguments.
 
-  * `search`, `-S` <text>|/<text>/:
+  * `search`, `-S` <tap>:
+    Display all formulae in a <tap>, even if not yet tapped.
+    <tap> is of the form <user>/<repo>, e.g. `brew search homebrew/dupes`.
```
Comment (mikemcquaid) [inline Library/Contributions/manpages/brew.1.md]: @davidfokkema Any thoughts?

**Comment by bfontaine** [comment]
cc @Homebrew/owners

File: Library/Contributions/manpages/brew.1.md (hunk 1 above)
Comment (davidfokkema) [inline Library/Contributions/manpages/brew.1.md]: This works for both tapped and untapped taps. Also, the substring search below works for untapped taps.

**Comment by mikemcquaid** [comment]
Feels like this should be part of a different command e.g. `brew ls` or `brew tap-info` rather than `search`.

**Comment by davidfokkema** [comment]
Currently, ``brew search`` lists all available formulae. In this PR, ``brew search <tap>`` lists all available formulae of a (untapped) tap. ``brew search <query>`` searches all available formulae, and ``brew search <tap> <query>`` searches a (untapped) tap. Seems consistent to me.

**Comment by bfontaine** [comment]
This needs rebasing on top of master.

**Comment by bfontaine** [comment]
I feel like `brew search <tap>` shouldn’t be documented and should instead be an edge-case of `brew search <tap> <query>` where the query is empty.

**Comment by mikemcquaid** [comment]
> I feel like brew search <tap> shouldn’t be documented as it’s an edge-case of brew search <tap> <query> where the query is empty.

Agreed :+1:

**Comment by davidfokkema** [comment]
Is there still interest, then? By now, it might need some serious work. If the interest is minimal, we could also just close this PR.

**Comment by mikemcquaid** [comment]
@davidfokkema If you're still interested I think the "search restricted to a specific tap" is neat. If you're not: don't worry about it!

**Comment by davidfokkema** [comment]
Due to time constraints and the fact that a grep will limit search to a tapped tap so provides some of the functionality, I'm closing this for now. Don't want to distract from other PRs.

**Pull request closed without merging**